File: projects/neptune/src/main/java/io/fairspace/neptune/config/RabbitMqProperties.java
package io.fairspace.neptune.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix="app.rabbitmq")
@Data
public class RabbitMqProperties {
    private boolean enabled = true;
    private RabbitMqTopology topology;

    @Data
    static class RabbitMqTopology {
        private RabbitMqCollectionsTopology collections;
    }

    @Data
    static class RabbitMqCollectionsTopology {
        private String exchange;
    }
}

File: src/main/kotlin/io/fairspace/ceres/events/config/RabbitMqProperties.kt
package io.fairspace.ceres.events.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component

@Configuration
@ConfigurationProperties(prefix="app.rabbitmq")
class RabbitMqProperties {
    var enabled = true
    var topology = RabbitMqTopology()

    data class RabbitMqTopology(var storage: RabbitMqStorageTopology = RabbitMqStorageTopology())
    data class RabbitMqStorageTopology(
            var exchange: String = "storage",
            var queues: Map<String, String> = emptyMap()
    )
}


